### ISSUE TITLE
AWS::EC2::LaunchTemplate.IamInstanceProfile not validated correctly

### ISSUE THREAD

**Issue by trav-c** [aws-cloudformation/cfn-python-lint#1552]
*cfn-lint version: (`cfn-lint --version`)*
cfn-lint 0.32.1

*Description of issue.*
When specifying an AWS::EC2::LaunchTemplate cfn-lint does not flag specifying the IamInstanceProfile property as a string value as an error, but it is actually required to be an object consisting of either an Arn or Name only.

Example of a faulty value that is not detected by cfn-lint 0.32.1
```
    ECSLaunchTemplate:
        Type: AWS::EC2::LaunchTemplate
        Properties:
            LaunchTemplateName: 'ECS-LaunchTemplate'
            LaunchTemplateData:
                IamInstanceProfile: !Ref EC2InstanceProfile
                ImageId: !GetAtt AMI.Outputs.ImageId
                InstanceInitiatedShutdownBehavior: terminate
                InstanceType: t3a.xlarge
```

Error from CloudFormation at runtime
Property validation failure: [Value of property {/LaunchTemplateData/IamInstanceProfile} does not match type {Object}]

Example of one correct form of IamInstanceProfile (It can also be specified by Arn instead of Name)
```
    ECSLaunchTemplate:
        Type: AWS::EC2::LaunchTemplate
        Properties:
            LaunchTemplateName: 'ECS-LaunchTemplate'
            LaunchTemplateData:
                IamInstanceProfile:
                    Name: !Ref EC2InstanceProfile
                ImageId: !GetAtt AMI.Outputs.ImageId
                InstanceInitiatedShutdownBehavior: terminate
                InstanceType: t3a.xlarge
```

*See References*
* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-iaminstanceprofile
* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LaunchTemplateIamInstanceProfileSpecificationRequest.html

**Comment by kddejong** [comment]
@trav-c when I run that template on v0.32.1 I'm getting.  What is the Ref to in your scenario?

```
E3002 Invalid Property Resources/ECSLaunchTemplate/Properties/LaunchTemplateData/IamInstanceProfile/Ref
local/test6.yaml:7:9
```

**Comment by trav-c** [comment]
The Ref was to a resource of type AWS::IAM::InstanceProfile

**Comment by kddejong** [comment]
Looks like this has been resolved. I know get
```
E3002 Invalid Property Resources/ECSLaunchTemplate/Properties/LaunchTemplateData/IamInstanceProfile/Ref
```
